lib/hanami/entity.rb

```diff
@@ -61,238 +51,35 @@ module Hanami
   #
   # @see Hanami::Repository
   module Entity
-    # Inject the public API into the hosting class.
-    #
-    # @since 0.1.0
-    #
-    # @example With Object
-    #   require 'hanami/model'
-    #
-    #   class User
-    #     include Hanami::Entity
-    #   end
-    #
-    # @example With Struct
-    #   require 'hanami/model'
-    #
-    #   User = Struct.new(:id, :name) do
-    #     include Hanami::Entity
-    #   end
-    def self.included(base)
-      base.class_eval do
-        extend ClassMethods
-        attributes :id
-      end
+    def initialize(attributes = nil)
+      @attributes = Utils::Hash.new((attributes || {}).dup).symbolize!
```

Comment:
Yes, that ensures that an entity instance will always have an encapsulated hash.